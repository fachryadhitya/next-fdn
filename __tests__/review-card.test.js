import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ReviewCard from '../components/review-card';

const props = {
  user: 'atikaxr',
  profile: ['Oily', 'Medium Dark', 'Warm', '19 - 24'],
  product: {
    image: 'https://static.femaledaily.com/dyn/480/images/prod-pics/product_1525323944_Tonymoly_D_800x800.jpg',
    name: 'Tony Moly',
    desc: 'Delight Tony Tint'
  },
  star: 3,
  comment:
    'liptint pertamaku haha dengan harga yang affordable udah bisa membuat korean look di wajahku (ga cocok si haha), cairrr, mudah kering jadi'
};

describe('Product Card Component', () => {
  it('Render successfully', () => {
    const { container } = render(<ReviewCard />);
    expect(container).toMatchSnapshot();
  });

  it('Render component with custom props', () => {
    const { getByText, getByAltText } = render(<ReviewCard data={props} />);
    expect(getByText(props.user)).toBeInTheDocument();
    expect(getByText(props.product.desc)).toBeInTheDocument();
    expect(getByText(props.product.name)).toBeInTheDocument();
    const image = getByAltText('product picture');
    expect(image).toHaveAttribute('src', props.product.image);
  });

  it('shows comment correctly with state', async () => {
    const { getByText } = render(<ReviewCard data={props} />);
    const desc = getByText(props.comment);
    expect(desc).toHaveClass('line-clamp-3');
    const button = screen.getByRole('button', { name: 'show more' });
    fireEvent.click(button);
    const txt = await screen.findByText(props.comment);
    expect(txt).not.toHaveClass('line-clamp-3');
  });
});
