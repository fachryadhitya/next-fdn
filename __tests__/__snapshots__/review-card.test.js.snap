// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Card Component Render successfully 1`] = `
<div>
  <div
    class="md:w-1/2 border-2 rounded-md p-4 relative"
  >
    <div
      class="flex border-b-2 py-2 mb-2"
    >
      <img
        alt="product picture"
        class="max-w-full w-20 h-20 rounded-lg object-center object-cover"
        src="/tony_moly.jpeg"
      />
      <div>
        <h3
          class="title-font font-bold uppercase"
        >
           
           
        </h3>
        <h4
          class="text-md text-gray-700"
        >
           
           
        </h4>
      </div>
    </div>
    <div
      class="flex justify-between items-center mb-2"
    >
      <span
        class="inline-flex"
      />
      <span
        class="text-sm text-gray-400"
      >
        2 hours ago
      </span>
    </div>
    <p
      class="line-clamp-3"
    />
    <button
      class="text-red-400"
      type="button"
    >
      show more
    </button>
    <div
      class="w-10 h-10 mx-auto items-center justify-center mb-4 absolute bottom-0 -mb-8"
    >
      <img
        alt="editor profile"
        class="w-24 h-auto rounded-full object-center object-cover"
        src="https://image.femaledaily.com/dyn/80/images/user-pics/8cfa98979828b03949df998669bcca01.jpg"
      />
      <div
        class="text-black w-40 max-w-full"
      >
        <p>
           
        </p>
      </div>
    </div>
  </div>
</div>
`;
